### DISCUSSION TITLE
Automatically locking database when hardware token is removed

### DISCUSSION THREAD

**Discussion by TitanFoxX** [keepassxreboot/keepassxc#6820] [Ideas]
Hi all,

I connected my YubiKey 5 to my KeePassXC database. It would be nice that the database would be locked automatically when I pull the hardware token from the USB port. Is this a feature that would be possible ?

Sincerely,
Ruud

**Comment by droidmonkey** [comment]
That might cause problems since we need the token to save the database. If you had unsaved changes or non data changes then you would have to plug in your token again to save. The database would be left unlocked during that time or changes would be lost. This would also require us to continually poll to see of the specific usb token was plugged in, we track them by serial number, which is not a simple operation and does have false negatives sometimes. 

In all, I don't think it's a good idea based on my work with the yubikey interface.
